crates/services/p2p/src/p2p_service.rs

```diff
@@ -1036,6 +1042,63 @@ mod tests {
         stop_sender.send(()).unwrap();
     }
 
+    #[tokio::test]
+    #[instrument]
+    async fn our_node_in_reserved_nodes() {
+        let mut retries = 10;
+        // We use bind to get a random port for the node to listen on.
+        // We use a loop because the port might be taken between the time we drop the listener and it's used by libp2p.
+        // We don't use `build_service_from_config` because we want to drop the tcp_listener at the last moment.
+        // and we don't want to `.unwrap()` if the port is taken but we prefer retry.
+        let mut node = loop {
+            let tcp_listener = TcpListener::bind("127.0.0.1:0").unwrap();
```

Comment:
Maybe it is somehow possible pass the binding into the `FuelP2PService` and use in the p2p?

Reply:
I tried to find a method to give something else than a `MultiAddr` and for instance a listener. Didn't found any. If you find one I will happy to use it